JavaScript/wh7_tech skills/hw_m6/js/JS_timer.js

```diff
@@ -0,0 +1,132 @@
+// цифры таймера
+let hourElement = document.querySelector('.hour')
+let minuteElement = document.querySelector('.minute')
+let secondElement = document.querySelector('.second')
+let millisecondElement = document.querySelector('.millisecond')
+
+// Кнопки
+let startButton = document.querySelector('.start')
+let splitButton = document.querySelector('.split')
+let stopButton = document.querySelector('.stop')
+let resetButton = document.querySelector('.reset')
+
+//доп переменные
+let hour = 0;
+let minute = 0;
+let second = 0;
+let millisecond = 0;
+let interval;
+
+//задачи кнопок при нажатии
+startButton.addEventListener('click', () => {
+  if (startButton.innerText === 'start') {
+    clearInterval(interval)
+    interval = setInterval(startTimer, 10)
+    startButton.innerText = `pause`
+    startButton.style.backgroundColor = `#6780D7`
+  } else if(startButton.innerText === 'pause') {
+   	clearInterval(interval)
+    startButton.innerText = `start`
+    startButton.style.backgroundColor = `#329F35`
+	}
+})
+
+splitButton.addEventListener('click', () => {
+  let results = document.querySelector('.results')
+  let addResult = document.createElement('div')
+  addResult.classList.add('split-result')
+  addResult.innerText = `Split: ${hour}:${minute}:${second}:${millisecond}`
+  results.appendChild(addResult)
+})
+
+stopButton.addEventListener('click', () => {
+  let results = document.querySelector('.results')
+  let addResult = document.createElement('div')
+  addResult.classList.add('stop-result')
+  addResult.style.color = '#F13C20'
+  clearInterval(interval)
+  addResult.innerText = `Stop: ${hour}:${minute}:${second}:${millisecond}`
+  results.appendChild(addResult)
+  clearStart()
+})
+
+resetButton.addEventListener('click', () => {
+  clearInterval(interval)
+  clearTimer()
+  clearResults('.results')
+  clearStart()
+})
+
+function clearStart() {
+	if(startButton.innerText === 'pause') {
+    startButton.innerText = `start`
+    startButton.style.backgroundColor = `#1F6521`
+	}
+}
+
+function clearTimer() {
+  hour = 0
+  minute = 0
+  second = 0
+  millisecond = 0
+  hourElement.textContent = '00'
+  minuteElement.textContent = '00'
+  secondElement.textContent = '00'
+  millisecondElement.textContent = '00'
+}
+
+function clearResults(element) {
+  document.querySelector(element).innerHTML = "";
+}
+
+
+//счет таймера
+function startTimer() {
+  millisecond++
+
+  //milliseconds
+  if (millisecond < 9) {
+    millisecondElement.innerText = '0' + millisecond
+  } else if (millisecond > 9) {
+    millisecondElement.innerText = millisecond
+  }
```

Comment:
this code duplicated few times, can be moved to function
function decimalPlaces(html, value) {
  if (value < 9) {
    html.innerText = '0' + value;
  } else if (value> 9) {
    html.innerText = value
  }
}